packages/affiliates/libs/affiliates.js

```diff
@@ -51,16 +51,29 @@ const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko, synthPrice
     );
   }
   // Gets the value of collateral currency in USD.
-  async function getCollateralPriceHistory(address, currency = "usd", start, end) {
+  async function getCoingeckoPriceHistory(address, currency = "usd", start, end) {
     const result = await coingecko.getHistoricContractPrices(address, currency, start, end);
     return Prices(result);
   }
 
+  // the fallback requires a different value calculation, so this is probably the cleanest way to "switch"
+  // between value by and value by usd. The function to calculate value is passed along with the data.
+  async function getSyntheticPriceHistoryWithFallback(address,start,end, syntheticAddress){
+    try{
+      // this will throw if the feed isnt available (among other errors probably)
+      return [await getSyntheticPriceHistory(address,start,end), calculateValue]
+    }catch(err){
+      // this will need to do a lookup based on synthetic token address not emp
```

Comment:
Is coingecko less reliable than what `getSyntheticPriceHistory` returns? Just want to check that the fallback is a worst option than the default

Reply:
its really more to offer more options for getting the value of the synthetic. originally we use cryptowatch which returns the synthetic value compared to its collateral, but ugas does not have this feed. In this case it has a coingecko feed with returns synthetic value compared to usd. So now we can basically offer support for a wider variety of synthetic types.

Reply:
yup. Just to add a bit more context the `getSyntheticPriceHistory` and cryptowatch feeds use the `financial-templates-lib`'s price feed which includes the `getHistoricalPricePeriods` which gives you all data over a large window. This method is not (yet) implemented in the uniswap feed. We _could_ add this method to the uniswap feed but this would not really solve the underlying problem that we cant support feeds that do not expose this `getHistoricalPricePeriods` method publically. Therefore, by adding the coingecko feed, we can support a much wider range; any product that is listed on coingecko.